pkg/config/identity/kubernetes_auth.go

```diff
@@ -0,0 +1,76 @@
+package identity
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/3scale-labs/authorino/pkg/common"
+	"github.com/3scale-labs/authorino/pkg/common/auth_credentials"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	authv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	authenticationv1 "k8s.io/client-go/kubernetes/typed/authentication/v1"
+	"k8s.io/client-go/rest"
+)
+
+type kubernetesAuthDetails struct {
+	authenticator authenticationv1.AuthenticationV1Interface
+	serviceToken  string
+}
+
+type KubernetesAuth struct {
+	auth_credentials.AuthCredentials
+	kubernetesAuthDetails
+}
+
+func NewKubernetesAuthIdentity(authCred auth_credentials.AuthCredentials) *KubernetesAuth {
+	config, _ := rest.InClusterConfig()
+	k8sClient, _ := kubernetes.NewForConfig(config)
```

Comment:
What are the chances of hitting a nasty error?

Reply:
The exact chance I wouldn't know, but it's possible. If somebody messes with some configs of the cluster, happens to override some cluster-reserved env vars within Authorino container, etc...

I will add a safe-guard clause here so it doesn't panic in line 37.

Thanks.

Reply:
Done